src/lowering/desugar.ml

```diff
@@ -657,6 +657,21 @@ and text_dotE proj e =
     | "chars" -> call "@text_chars" [] [T.iter_obj T.char]
     |  _ -> assert false
 
+and let_else_switch p e f = 
+  let e' = exp e in
+  let p' = pat p in
+  let f' = exp f in
+  I.{ 
+    it = I.SwitchE(
+      exp e,
+      [
+        { it = { pat = p'; exp = e' }; at = e'.at; note = () };
+        { it = { pat = { it = WildP; at = f'.at; note = p'.note }; exp = f' }; at = f'.at ; note = () }
```

Comment:
normally you could use `cases` here if the input is on the `S` (`Source`) side...